src/Stratis.Features.Collateral/CheckCollateralFullValidationRule.cs

```diff
@@ -48,24 +54,64 @@ public override Task RunAsync(RuleContext context)
             if (commitmentHeight == null)
             {
                 // We return here as it is CheckCollateralCommitmentHeightRule's responsibility to perform this check.
                 this.Logger.LogTrace("(-)SKIPPED_AS_COLLATERAL_COMMITMENT_HEIGHT_MISSING]");
                 return Task.CompletedTask;
             }
 
-            this.Logger.LogDebug("Commitment is: {0}.", commitmentHeight);
+            this.Logger.LogDebug("Commitment is: {0}. Magic is: {1}", commitmentHeight, magic);
 
-            // TODO: Both this and CollateralPoAMiner are using this chain's MaxReorg instead of the Counter chain's MaxReorg. Beware: fixing requires fork.
+            // TODO: The code contained in the following "if" can be removed after most nodes have switched their mainchain to Strax.
 
+            // Strategy:
+            // 1. I'm a Cirrus miner on STRAX. If the block's miner is also on STRAX then check the collateral. Pass or Fail as appropriate.
+            // 2. The block miner is on STRAT. If most nodes were on STRAT(prev round) then they will check the rule. Pass the rule.
+            // 3. The miner is on STRAT and most nodes were on STRAX(prev round).Fail the rule.
+
+            // 1. If the block miner is on STRAX then skip this code and go check the collateral.
+            Network counterChainNetwork = this.fullNode.NodeService<CounterChainNetworkWrapper>().CounterChainNetwork;
+            if (this.network.Name.StartsWith("Cirrus") && magic != counterChainNetwork.Magic)
+            {
+                // 2. The block miner is on STRAT.
+                IConsensusManager consensusManager = this.fullNode.NodeService<IConsensusManager>();
+                int memberCount = 0;
+                int membersOnStratis = 0;
+                uint targetSpacing = this.slotsManager.GetRoundLengthSeconds();
+                ChainedHeader chainedHeader = context.ValidationContext.ChainedHeaderToValidate;
+                do
+                {
+                    chainedHeader = chainedHeader.Previous;
+
+                    if (chainedHeader.Block == null)
+                        chainedHeader.Block = consensusManager.GetBlockData(chainedHeader.HashBlock).Block;
+
+                    (int? commitmentHeight2, uint? magic2) = commitmentHeightEncoder.DecodeCommitmentHeight(chainedHeader.Block.Transactions.First());
+                    if (commitmentHeight2 == null)
+                        continue;
+
+                    if (magic2 != counterChainNetwork.Magic)
+                        membersOnStratis++;
+
+                    memberCount++;
+                } while ((chainedHeader.Block.Header.Time + targetSpacing) > context.ValidationContext.BlockToValidate.Header.Time);
+
+                // If most nodes were on STRAT(prev round) then they will check the rule. Pass the rule.
+                if (membersOnStratis * 2 >= memberCount)
```

Comment:
So if the block producers in this round is split 50/50 then we pass? 🤔 I'm wondering whether or not we should already at this point start validation?

Reply:
If the current block being validated would not be in the round then the round could in some circumstance (no collateral commitment height in any sampled block) have memberCount = 0 and membersOnStratis = 0. In such a scenario one would not want to fail the rule. I've made changes above and here to guarantee a memberCount of at least 1 and hence this condition could not be changed.